web/src/compose_call_ui.ts

```diff
@@ -10,38 +11,53 @@ import {current_user, realm} from "./state_data";
 import * as ui_report from "./ui_report";
 import * as util from "./util";
 
-export function update_audio_and_video_chat_button_display() {
+const call_response_schema = z.object({
+    msg: z.string(),
+    result: z.string(),
+    url: z.string(),
+});
+
+export function update_audio_and_video_chat_button_display(): void {
     update_audio_chat_button_display();
     update_video_chat_button_display();
 }
 
-export function update_video_chat_button_display() {
+export function update_video_chat_button_display(): void {
     const show_video_chat_button = compose_call.compute_show_video_chat_button();
     $(".compose-control-buttons-container .video_link").toggle(show_video_chat_button);
     $(".message-edit-feature-group .video_link").toggle(show_video_chat_button);
 }
 
-export function update_audio_chat_button_display() {
+export function update_audio_chat_button_display(): void {
     const show_audio_chat_button = compose_call.compute_show_audio_chat_button();
     $(".compose-control-buttons-container .audio_link").toggle(show_audio_chat_button);
     $(".message-edit-feature-group .audio_link").toggle(show_audio_chat_button);
 }
 
-function insert_video_call_url(url, target_textarea) {
+function insert_video_call_url(
+    url: string,
+    target_textarea: JQuery<HTMLTextAreaElement> | undefined,
```

Comment:
Under what circumstances can `target_textarea` be undefined`? I'm suspicious that you're changing the types due to a buggy test; if that's the case, one should fix the test to call the function properly.

`insert_syntax_and_focus` expects a `JQuery<HTMLTextAreaElement>`, never `undefined`, BTW.

Reply:
This can be `undefined` when we are inserting the video call link in the text box while composing a message (I am able to reproduce this manually). We only pass `target_textarea` when we want to insert the video call in the edit form.

Yes `insert_syntax_and_focus` expects a `JQuery<HTMLTextAreaElement>` which defaults to `$<HTMLTextAreaElement>("textarea#compose-textarea")` when nothing is passed into it.

Reply:
OK. Might make sense to refactor to have `target_textarea` be the compose one rather than implied via `undefined`, but that can be a follow-up.